spectral_cube/tests/test_stokes.py

```diff
@@ -0,0 +1,138 @@
+                           Q=self.data[1],
+                           U=self.data[2],
+                           V=self.data[3])
+        with pytest.raises(TypeError) as exc:
+            cube = StokesSpectralCube(stokes_data, self.wcs)
+        assert exc.value.args[0] == "stokes_data should be a dictionary of SpectralCube objects"
+
+    def test_direct_init_invalid_shape(self):
+        stokes_data = dict(I=SpectralCube(np.ones((6, 2, 30)), self.wcs),
+                           Q=SpectralCube(self.data[1], self.wcs),
+                           U=SpectralCube(self.data[2], self.wcs),
+                           V=SpectralCube(self.data[3], self.wcs))
+        with pytest.raises(ValueError) as exc:
+            cube = StokesSpectralCube(stokes_data, self.wcs)
+        assert exc.value.args[0] == "All spectral cubes shoul have the same shape"
+
+
+    @pytest.mark.parametrize('component', ('I', 'Q', 'U', 'V', 'RR', 'RL', 'LR', 'LL'))
+    def test_valid_component_name(self, component):
+        stokes_data = {component: SpectralCube(self.data[0], self.wcs)}
+        cube = StokesSpectralCube(stokes_data)
+
+    @pytest.mark.parametrize('component', ('A', 'B', 'IQUV'))
+    def test_invalid_component_name(self, component):
+        stokes_data = {component: SpectralCube(self.data[0], self.wcs)}
+        with pytest.raises(ValueError) as exc:
+            cube = StokesSpectralCube(stokes_data, self.wcs)
+        assert exc.value.args[0] == "Invalid Stokes component: {0} - should be one of I, Q, U, V, RR, LL, RL, LR".format(component)
+
+    def test_invalid_wcs(self):
+        wcs2 = WCS(naxis=3)
+        wcs2.wcs.ctype = ['GLON-CAR', 'GLAT-CAR', 'FREQ']
+        stokes_data = dict(I=SpectralCube(self.data[0], self.wcs),
+                           Q=SpectralCube(self.data[1], wcs2))
+        with pytest.raises(ValueError) as exc:
+            cube = StokesSpectralCube(stokes_data)
+        assert exc.value.args[0] == "All spectral cubes in stokes_data should have the same WCS"
+
+    def test_attributes(self):
+        stokes_data = dict(I=SpectralCube(self.data[0], self.wcs),
+                           Q=SpectralCube(self.data[1], self.wcs),
+                           U=SpectralCube(self.data[2], self.wcs),
+                           V=SpectralCube(self.data[3], self.wcs))
+        cube = StokesSpectralCube(stokes_data)
+        assert_allclose(cube.I.unmasked_data[...], 0)
+        assert_allclose(cube.Q.unmasked_data[...], 1)
+        assert_allclose(cube.U.unmasked_data[...], 2)
+        assert_allclose(cube.V.unmasked_data[...], 3)
+
+    def test_dir(self):
+        stokes_data = dict(I=SpectralCube(self.data[0], self.wcs),
+                           Q=SpectralCube(self.data[1], self.wcs),
+                           U=SpectralCube(self.data[2], self.wcs))
+        cube = StokesSpectralCube(stokes_data)
+        for stokes in 'IQU':
+            assert stokes in cube.__dir__()
+        assert 'V' not in cube.__dir__()
```

Comment:
related to above, `assert 'mask' in dir(cube)`.  I think it's better to test the public `dir(x)` rather than the private `x.__dir__()`?  They're equivalent, but... somehow, `dir(x)` feels more right to me.  I can't justify that.